base/collection.go

```diff
@@ -621,14 +624,19 @@ func (b *GocbV2Bucket) DefaultDataStore() sgbucket.DataStore {
 }
 
 // NamedDataStore returns a collection on a bucket within the given scope and collection.
-func (b *GocbV2Bucket) NamedDataStore(name sgbucket.DataStoreName) sgbucket.DataStore {
+func (b *GocbV2Bucket) NamedDataStore(name sgbucket.DataStoreName) (sgbucket.DataStore, error) {
+	err := WaitForNoError(func() error {
+		_, err := NewCollection(
+			b, b.bucket.Scope(name.ScopeName()).Collection(name.CollectionName()))
+		return err
+	})
+	if err != nil {
+		return nil, fmt.Errorf("attempting to create/update database with a scope/collection that is not found: %w", ErrAuthError)
+	}
```

Comment:
I am not sure why this function needs to call `NewCollection` twice?

Also not sure the error retry is best suited here specifically. I'd expected it up in `_getOrAddDatabaseFromConfig` so that NamedDataStore itself is retried before calling `WaitUntilDataStoreExists`

Maybe those two can be combined into a new function if it's a common pattern to do?

Reply:
If the retry loop is done before the `WaitUntilDataStoreExists` in `_getOrAddDatabaseFromConfig`, is it not better to call `NamedDataStore` after `WaitUntilDataStoreExists` since the retry loop would be doing the same thing as `WaitUntilDataStoreExists`?

Reply:
https://github.com/couchbase/sync_gateway/pull/6018/commits/6e9cef6ef7089a499dd1e20509f45af572390485